tests/lapi/coroutine_test.lua

```diff
@@ -0,0 +1,177 @@
+https://www.lua.org/bugs.html#5.3.0-3
+
+Resuming the running coroutine makes it unyieldable,
+https://www.lua.org/bugs.html#5.2.2-8
+
+pcall may not restore previous error function when inside coroutines,
+https://www.lua.org/bugs.html#5.2.1-2
+
+Wrong handling of nCcalls in coroutines,
+https://www.lua.org/bugs.html#5.2.0-4
+
+coroutine.resume pushes element without ensuring stack size,
+https://www.lua.org/bugs.html#5.1.3-2
+
+Recursive coroutines may overflow C stack,
+https://www.lua.org/bugs.html#5.1.2-4
+
+Stand-alone interpreter shows incorrect error message when the
+"message" is a coroutine,
+https://www.lua.org/bugs.html#5.1.2-12
+
+Debug hooks may get wrong when mixed with coroutines,
+https://www.lua.org/bugs.html#5.1-7
+
+Values held in open upvalues of suspended threads may be
+incorrectly collected,
+https://www.lua.org/bugs.html#5.0.2-3
+
+Attempt to resume a running coroutine crashes Lua,
+https://www.lua.org/bugs.html#5.0-2
+
+debug.getlocal on a coroutine suspended in a hook can crash the interpreter,
+https://www.lua.org/bugs.html#5.3.0-2
+
+debug.sethook/gethook may overflow the thread's stack,
+https://www.lua.org/bugs.html#5.1.2-13
+
+Memory hoarding when creating Lua hooks for coroutines,
+https://www.lua.org/bugs.html#5.2.0-1
+
+Synopsis:
+
+coroutine.close(co)
+coroutine.create(f)
+coroutine.isyieldable([co])
+coroutine.resume(co [, val1, ...])
+coroutine.running()
+coroutine.status(co)
+coroutine.wrap(f)
+coroutine.yield(...)
+]]
+
+local luzer = require("luzer")
+local test_lib = require("lib")
+
+local CORO_OBJECTS
+
+-- Possible coroutine statuses, described in Lua 5.1 Reference Manual,
+-- https://www.lua.org/manual/5.4/manual.html#6.2
+local CORO_STATUS = {
+    DEAD = "dead",
+    NORMAL = "normal",
+    RUNNING = "running",
+    SUSPENDED = "suspended",
+}
+
+local CORO_ACTION_NAME = {
+    "close",
+    "create",
+    "resume",
+    "yield",
+}
+
+  -- Forward declaration.
+local coro_function
+
+local function hook_func(_event)
+    -- Accessing Locals,
+    -- https://www.lua.org/pil/23.1.1.html.
+    local level = 2
+    local i = 1
+    while true do
+        local name, _ = debug.getlocal(level, i)
+        if not name then break end
+        i = i + 1
+    end
+    -- Accessing Upvalues,
+    -- https://www.lua.org/pil/23.1.2.html
+    local func = debug.getinfo(level).func
+    i = 1
+    while true do
+        local name, _ = debug.getupvalue(func, i)
+        if not name then break end
+        i = i + 1
+    end
+end
+
+local function sethook(co, fdp)
+    local set_hook = fdp:consume_boolean()
+    local hook_args = {}
+    if not set_hook then
+        return
+    end
+    table.insert(hook_args, hook_func)
+    table.insert(hook_args, fdp:oneof({"c", "r", "l"}))
+    debug.sethook(co, unpack(hook_args))
+end
+
+local function coro_random_action(fdp, coro_max_number)
+    local action = fdp:oneof(CORO_ACTION_NAME)
+    if action == "create" or #CORO_OBJECTS < coro_max_number then
```

Comment:
Should it be `and` instead of `or` to properly limit `CORO_OBJECTS`?

If yes, we need an additional an assertion check (due to the assumption that `TestOneInput()` will initialize at least one coroutine) that `CORO_OBJECTS` isn't empty table.

Reply:
Right, it was not force-pushed, sorry.